WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -43,13 +43,39 @@ public class CoinJoinClient
 		public KeyManager Keymanager { get; }
 		private RoundStateUpdater RoundStatusUpdater { get; }
 
-		public async Task StartCoinJoinAsync(CancellationToken cancellationToken, IEnumerable<Money>? forcedOutputDenominations = null)
+		public async Task<bool> StartCoinJoinAsync(CancellationToken cancellationToken)
+		{
+			var currentRoundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState => roundState.Phase == Phase.InputRegistration, cancellationToken).ConfigureAwait(false);
+
+			// This should be roughly log(#inputs), it could be set slightly
+			// higher if more inputs are observed but that involves trusting the
+			// coordinator with those values. Therefore, conservatively set this
+			// so that a maximum of 5 blame rounds are executed.
+			// FIXME should smaller rounds abort earlier?
```

Comment:
I would go with a fixed number here. What happens after 5 failed blame rounds? The client will go to another round after? Can it go to another round with the same Coins? What is the cost of complete sabotage of one round?

Reply:
The client should stop trying to register those coins for a while, I think. Need to work out details in #6043